File: app/src/main/java/by/heapix/proslau/heapixtelegram/View/fragments/ChatFragment.java
package by.heapix.proslau.heapixtelegram.view.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.*;
import by.heapix.proslau.heapixtelegram.R;

import java.util.ArrayList;


public class ChatFragment extends Fragment {
    private TextView tvUsername;
    private TextView tvLastMessage;
    private TextView tvDate;

    final String ATTRIBUTE_USERNAME = "username";
    final String ATTRIBUTE_LAST_MESSAGE = "lastMessage";
    final String ATTRIBUTE_DATE = "date";

    String[] users = {"Alex", "Pedro", "Nick"};
    String[] lastMessages = {"ok", "lol", "kek"};
    String[] dates = {"Mon", "Sun", "Thu"};

    RecyclerView recyclerView;
    ArrayAdapter adapter;
    ArrayList<String[]> data;
    String[] strings;


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.chat_fragment, container, false);

        LinearLayout linLayout = (LinearLayout) view.findViewById(R.id.lin_layout_chat);

        LayoutInflater ltInflater = getLayoutInflater();

        for (int i = users.length - 1; i >= 0; i--) {
            View item = ltInflater.inflate(R.layout.chat_item, linLayout, false);
            tvUsername = (TextView) item.findViewById(R.id.tv_chat_item_name);
            tvUsername.setText(String.valueOf(users[i]));
            tvLastMessage = (TextView) item.findViewById(R.id.tv_chat_item_last_message);
            tvLastMessage.setText(String.valueOf(lastMessages[i]));
            tvDate = (TextView) item.findViewById(R.id.tv_chat_date);
            tvDate.setText(String.valueOf(dates[i]));
            item.getLayoutParams().width = ViewGroup.LayoutParams.MATCH_PARENT;
            linLayout.addView(item);
        }

        /*data = new ArrayList<>();
        for(int i = users.length - 1; i >= 0; i--){
            strings = new String[]{users[i], lastMessages[i], dates[i]};
            data.add(strings);
        }

        String[] from = {ATTRIBUTE_USERNAME, ATTRIBUTE_LAST_MESSAGE, ATTRIBUTE_DATE};
        int[] to = {R.id.tv_chat_item_name, R.id.tv_chat_item_last_message, R.id.tv_chat_date};

        adapter = new ArrayAdapter(view.getContext(), data, R.layout.chat_item, from, to);*/


        return view;
    }


}

File: app/src/main/java/by/heapix/proslau/heapixtelegram/view/fragments/ChatFragment.kt
package by.heapix.proslau.heapixtelegram.view.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.SearchView
import android.support.v7.widget.Toolbar
import android.view.*
import by.heapix.proslau.heapixtelegram.R
import by.heapix.proslau.heapixtelegram.controller.adapters.ChatListRecyclerViewAdapter
import by.heapix.proslau.heapixtelegram.model.ChatListItem
import by.heapix.proslau.heapixtelegram.model.ChatListItem.Companion.initChatList


class ChatFragment : Fragment(), SearchView.OnQueryTextListener {

    internal lateinit var view: View
    lateinit var adapter: ChatListRecyclerViewAdapter
    var chatList : ArrayList<ChatListItem> = ArrayList()


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        view = inflater.inflate(R.layout.chat_list_fragment, container, false)

        initImageBitmaps()

        val toolbar = view.findViewById<View>(R.id.chat_toolbar) as Toolbar
        (activity as AppCompatActivity).setSupportActionBar(toolbar)
        toolbar.inflateMenu(R.menu.menu_toolbar)
        toolbar.title = "Chat"
        val menuItem = toolbar.menu!!.findItem(R.id.action_search)
        val searchView : android.support.v7.widget.SearchView = menuItem.actionView as android.support.v7.widget.SearchView
        searchView.setOnQueryTextListener(this)
        searchView.queryHint = "Search"

        return view
    }

    override fun onQueryTextSubmit(p0: String?): Boolean {
        return false
    }

    override fun onQueryTextChange(p0: String?): Boolean {
        val userInput = p0!!.toLowerCase()
        val newList = ArrayList<ChatListItem>()
        chatList.forEach {
            if(it.name.toLowerCase().contains(userInput) || it.lastMessage.toLowerCase().contains(userInput)){
                newList.add(it)
            }
        }
        adapter.updateList(newList)
        return true
    }

    private fun initImageBitmaps() {
        chatList = initChatList()
        initChatListRecyclerView()
    }

    private fun initChatListRecyclerView() {
        adapter = ChatListRecyclerViewAdapter(chatList, context)
        val recyclerView = view.findViewById<RecyclerView>(R.id.chat_list_recycler_view)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(context)
    }
}
